Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoftWindowsFirewall.yaml

```diff
@@ -13,31 +13,164 @@ References:
+                  DstUserId = iff(isOutBound, RemoteUserID, ""),
+                  SrcUserId = iff(not(isOutBound), RemoteUserID, "")
+            };
+            //////////////////////////////////////////////////////
+            // this query extract the data from WindowsEvent table
+            //////////////////////////////////////////////////////
+            let WindowsFirewall_WindowsEvent=(){   
+                      WindowsEvent
+                      | where EventID between (5150 .. 5159)
+                      | extend 
+                          EventSeverity=tostring(EventData.Severity),
+                          LayerCode = tostring(EventData.LayerName),
+                          NetworkRuleNumber = tostring(EventData.FilterRTID),
+                          Protocol = toint(EventData.Protocol),
+                          DirectionCode = iff(EventID in (5154, 5155, 5158, 5159), "%%14609",tostring(EventData.Direction))
+                      | extend iff(EventID in (5154, 5155, 5158, 5159), DirectionCode="%%14609",DirectionCode = tostring(EventData.Direction) )
+                      | lookup Directions on DirectionCode // need to fix this extend -> need to take the column name out
+                      | extend  SrcAppName = iff(isOutBound, tostring(EventData.Application), ""),
+                                DstAppName = iff(not(isOutBound), tostring(EventData.Application), ""),
+                                SrcIpAddr = iff(isOutBound, tostring(EventData.DestAddress), tostring(EventData.SourceAddress)),
+                                DstIpAddr = iff(not(isOutBound), tostring(EventData.SourceAddress), tostring(EventData.DestAddress)),
+                                SrcHostId = iff(isOutBound, tostring(EventData.RemoteMachineID), ""),
+                                DstHostId = iff(not(isOutBound), tostring(EventData.RemoteMachineID), ""),
+                                SrcPortNumber = iff(isOutBound, toint(EventData.DestPort), toint(EventData.SourcePort)),
+                                DstPortNumber = iff(not(isOutBound), toint(EventData.SourcePort), toint(EventData.DestPort)),
+                                SrcProcessId =  iff(isOutBound, toint(EventData.ProcessId), toint("")),
+                                DstProcessId =  iff(not(isOutBound), toint(EventData.ProcessId), toint("")),
+                                DstUserId = iff(isOutBound, tostring(EventData.RemoteUserID), ""),
+                                SrcUserId = iff(not(isOutBound), tostring(EventData.RemoteUserID), "")
+                  };
+            // Main query -> outputs both schemas as one normalized table
+            union isfuzzy=true WindowsFirewall_SecurityEvent, WindowsFirewall_WindowsEvent
+                | extend 
+                        DvcAction = iff(EventID in (5154, 5156, 5158), "Allow", "Deny"),
+                        DvcOs = 'Windows',
+                        DstAppType = "Process",
```

Comment:
In the future - I usually don't assign a type when there is no value. In your case, it is a lot more iff statements. A good reason to use union next time :-)